app/src/main/java/ch/epfl/sdp/appart/utils/serializers/UserSerializer.java

```diff
@@ -23,36 +22,41 @@
         docData.put(UserLayout.NAME, data.getName());
         docData.put(UserLayout.PHONE, data.getPhoneNumber());
         docData.put(UserLayout.PICTURE, data.getProfileImage());
+        docData.put(UserLayout.AD_IDS, data.getAdsIds());
+        docData.put(UserLayout.FAVORITE_IDS, new ArrayList<>(data.getFavoritesIds()));
         return docData;
     }
 
     @Override
     public User deserialize(String id, Map<String, Object> data) {
         AppUser user = new AppUser(id, (String) data.get(UserLayout.EMAIL));
 
-        Object rawAge = data.get(UserLayout.AGE);
-        if (rawAge != null) {
-            user.setAge((long) rawAge);
+        if (data.get(UserLayout.AGE) != null) {
+            user.setAge((long) data.get(UserLayout.AGE));
+        }
+
+        if (data.get(UserLayout.GENDER) != null) {
+            user.setGender((String) data.get(UserLayout.GENDER));
+        }
+
+        if (data.get(UserLayout.NAME) != null) {
+            user.setName((String) data.get(UserLayout.NAME));
         }
 
-        Object rawGender = data.get(UserLayout.GENDER);
-        if (rawGender != null) {
-            user.setGender((String) rawGender);
+        if (data.get(UserLayout.PHONE) != null) {
+            user.setPhoneNumber((String) data.get(UserLayout.PHONE));
         }
 
-        Object rawName = data.get(UserLayout.NAME);
-        if (rawName != null) {
-            user.setName((String) rawName);
+        if (data.get(UserLayout.PICTURE) != null) {
+            user.setProfileImage((String) data.get(UserLayout.PICTURE)); //WARNING WAS "profilePicture" before not matching our actual
         }
 
-        Object rawPhoneNumber = data.get(UserLayout.PHONE);
-        if (rawPhoneNumber != null) {
-            user.setPhoneNumber((String) rawPhoneNumber);
+        if (data.get(UserLayout.AD_IDS) !=  null) {
+            ((List<String>) data.get(UserLayout.AD_IDS)).forEach(user::addAdId);
         }
 
-        Object rawPfpRef = data.get(UserLayout.PICTURE);
-        if (rawPfpRef != null) {
-            user.setProfileImage((String) rawPfpRef); //WARNING WAS "profilePicture" before not matching our actual
+        if (data.get(UserLayout.FAVORITE_IDS) !=  null) {
+            ((List<String>) data.get(UserLayout.FAVORITE_IDS)).forEach(user::addFavorite);

```

Comment:
It's important to know, because if I got everything in the PR, the User does not contain the two lists and that could create issues for others working with those values. I don't have problems adding the necessary part of code for the local favorites logic, but I need to know if it was intended to not add that part of code here

Reply:
I don't really understand what you mean. Currently, users have two lists, one having the ids of all the ads he created, the other having the ids of the favorite ads of the user. Thus, it is normal for these to be (de)serialized in the userSerializer, isn't it ?

Reply:
My bad I didn't notice that you implemented the two lists as part of the document of the user, and not as a separate collection as we did for the ad and its photos references. You can ignore my comments